pyof/v0x04/controller2switch/multipart_reply.py

```diff
@@ -288,7 +290,21 @@ def __init__(self, length=None, table_id=None, duration_sec=None,
         self.cookie = cookie
         self.packet_count = packet_count
         self.byte_count = byte_count
-        self.match = match
+        self.match = match or Match()
+        self.instructions = instructions or ListOfInstruction()
+
+    def unpack(self, buff, offset=0):
+        """Unpack a binary message into this object's attributes.
+
+        Pass the correct length for list unpacking.
+
+        Args:
+            buff (bytes): Binary data package to be unpacked.
+            offset (int): Where to begin unpacking.
+        """
+        unpack_length = UBInt16()
+        unpack_length.unpack(buff, offset)
+        super().unpack(buff[:offset+unpack_length], offset)
```

Comment:
@renanrodrigo just one question here.

Python slicing is somewhat tricky sometimes. Have you tested this expression? The right slicing here is `[:offset+unpack_length]` or is it `[:offset+unpack_length+1]`?

Reply:
I tested and this seems right.
All other list implementations use this form (without the `+1`).
I'm writing tests for this structure to ensure future changes won't compromise things.